### DISCUSSION TITLE
Missing bootstrap class

### DISCUSSION THREAD

**Discussion by mhite** [thuliteio/doks#653] [Q&A] [answered]
Hello, apologies for the not-very-savvy frontend question.

I've got a site that looks fine when I view and launch it via

```npm run start```

When it gets deployed to Netlify, it appears a class is missing. Specifically, the "ratio-4x3" class that gets applied to a video `<div>`.

If you scroll to the very bottom of this page, there is supposed to be a video:

https://staging--distracted-edison-fcd4c8.netlify.app/docs/quickstart/

I think it is not displayed because the class applied has a height of 0.

It is supposed to look like this:

![image](https://user-images.githubusercontent.com/1464385/152420037-df0e4fc8-2225-4626-9f63-196bfa56d0b1.png)

I believe the major difference is that the local version is not minified bootstrap.js, whereas the content that gets deployed on netlify is.

Any tips on how to troubleshoot and fix further? I'm a backend guy futzing with frontend but trying to learn. Any pointers appreciated.

Pertinent markdown:

```
## Go DJ! 🎵

Start DJing! You'll see track transitions automatically update in the browser view. You should also see track transition messages log to the terminal.

{{< video ratio="4x3" attributes="controls autoplay muted" webm-src="videos/djctl_browser_demo.webm" mp4-src="videos/djctl_browser_demo.mp4" >}}
```

Also --

I tried removing `--minify` from the build command in the `package.json` file. That didn't fix anything, so it's [likely] not Hugo munching it?

```
"build": "exec-bin node_modules/.bin/hugo/hugo --gc --minify",
```

I'm not sure, but this looks like perhaps a similar issue?

https://github.com/h-enk/doks/issues/144

`npm run check` output:

```
$ npm run check

> doks-child-theme@0.3.5 precheck
> npm version

{
  'doks-child-theme': '0.3.5',
  npm: '8.3.1',
  node: '17.4.0',
  v8: '9.6.180.15-node.13',
  uv: '1.43.0',
  zlib: '1.2.11',
  brotli: '1.0.9',
  ares: '1.18.1',
  modules: '102',
  nghttp2: '1.46.0',
  napi: '8',
  llhttp: '6.0.4',
  openssl: '1.1.1m',
  cldr: '39.0',
  icu: '69.1',
  tz: '2021a',
  unicode: '13.0'
}

> doks-child-theme@0.3.5 check
> exec-bin node_modules/.bin/hugo/hugo version

hugo v0.88.1-5BC54738+extended darwin/amd64 BuildDate=2021-09-04T09:39:19Z VendorInfo=gohugoio
```

**Comment by h-enk** [comment, marked as answer]
> I'm not sure, but this looks like perhaps a similar issue?

That's correct (most likely — shortcode added classes cannot be seen by PostCSS).

So, try adding class `ratio-4x3` to the `safelist` section of  `./config/postcss.config.js`. That should do the trick.

**Reply by mhite** [reply]
That did it! Thank you!

Should I open a PR? We probably want:

- `ratio-1x1`
- `ratio-4x3`
- `ratio-16x9`
- `ratio-21x9`

Or something that just matches `ratio-`

**Reply by h-enk** [reply]
Good to hear that you got it working!

Yes, I guess we should add this to Doks — your PR is welcome.

I think using a [pattern](https://purgecss.com/safelisting.html#patterns) is a smart thing to do (if you get it working), otherwise just add the 4 separate classes. Thanks in advance!
